### PR TITLE
Feat/fetch fetch from API

### PR DESCRIPTION
**Pull request by D3nnis249**
Hello and thank you for your review. In this PR, we have exchanged our fake data for an Api fetch and added this fetch with a radius filter.

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -8,14 +8,34 @@ import SettingsPage from '../src/pages/SettingsPage';
...
+  const [position, setPosition] = useState(null);
   const [fuelValue, setFuelValue] = useState('e5');
   const [favoriteIDs, setFavoriteIDs] = useState(
     () => JSON.parse(localStorage.getItem('favoriteIDs')) ?? []
   );
 
+  const url = `https://api.tankentanken.de/gas-stations?latitude=${latitude}&longitude=${longitude}&radius=${apiRadius}`;
   const FuelInfo = fuelValue.charAt(0).toUpperCase() + fuelValue.slice(1);
 
+  //Calculate radius for API
```
Comment (mbosselmann) [inline client/src/App.jsx]: I think that these comments aren't really necessary as the names of your functions are good. Therefore, I would recommend you to remove them.

File: client/src/App.jsx [hunk 2]
```diff
@@ -8,14 +8,34 @@ import SettingsPage from '../src/pages/SettingsPage';
...
+  function calculateApiRadius(radius) {
```
Comment (mbosselmann) [inline client/src/App.jsx]: I think as name `calculateRadius` would be fine as this function could also be used for other purposes. :)

File: client/src/App.jsx [hunk 3]
```diff
@@ -25,36 +45,86 @@ function App() {
...
+  // Fetch Stations
+  function fetchStations() {
+    fetch(url, {
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${stationsApiKey}`,
+      },
+    })
+      .then(response => response.json())
+      .then(json => setStations(json));
```
Comment (mbosselmann) [inline client/src/App.jsx:54]: I would suggest you to add some error handling with `catch`

File: client/src/App.jsx [hunk 4]
```diff
@@ -8,14 +8,34 @@ import SettingsPage from '../src/pages/SettingsPage';
...
+  const [latitude, setLatitude] = useState(null);
+  const [longitude, setLongitude] = useState(null);
+  const [position, setPosition] = useState(null);
   const [fuelValue, setFuelValue] = useState('e5');
   const [favoriteIDs, setFavoriteIDs] = useState(
     () => JSON.parse(localStorage.getItem('favoriteIDs')) ?? []
   );
 
+  const url = `https://api.tankentanken.de/gas-stations?latitude=${latitude}&longitude=${longitude}&radius=${apiRadius}`;
   const FuelInfo = fuelValue.charAt(0).toUpperCase() + fuelValue.slice(1);
```
Comment (mbosselmann) [inline client/src/App.jsx]: As in React components names begin with a capital letter, you should not name this `const` `FuelInfo`, but `fuelInfo`.

File: client/src/components/Button.jsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+function Button({ myFunction, buttonTitle }) {
+  return <button onClick={myFunction}>{buttonTitle}</button>;
+}
+
+export default Button;
```
Comment (mbosselmann) [inline client/src/components/Button.jsx:7]: I like! 👍 But no stylings for this button? 🤔

**Review by mbosselmann — COMMENTED** [review]
You rock! 🪨 Left some remarks for you to consider. :)

File: client/src/App.jsx (hunk 4 above)
Comment (D3nnis249) [reply client/src/App.jsx]: Ups, thanks :D

File: client/src/components/Button.jsx (hunk 5 above)
Comment (D3nnis249) [reply client/src/components/Button.jsx:7]: We first took care of the functionality, and today we take care of the styling. 👍

File: client/src/App.jsx (hunk 3 above)
Comment (D3nnis249) [reply client/src/App.jsx:54]: Was discussed with Mareike 👍

File: client/src/components/GasStationList.jsx [hunk 6]
```diff
@@ -77,13 +89,17 @@ const CustomLink = styled(Link)`
 const PriceWrapper = styled.p`
   grid-area: price;
   align-self: center;
-  margin-right: 1em;
+  font-weight: 600;
+  /* margin-right: 1em; */
```
Comment (mbosselmann) [inline client/src/components/GasStationList.jsx]: Remove commented code please ;)

File: client/src/components/GasStationList.jsx [hunk 7]
```diff
@@ -99,18 +115,26 @@ const Adress = styled.p`
...
+  font-size: 2.25rem;
   color: #2196f3;
 `;
 
+const DistanceWrapper = styled.p`
+  grid-area: distance;
+  justify-self: end;
+  align-self: end;
+  font-weight: 600;
+  /* align-self: center; */
```
Comment (mbosselmann) [inline client/src/components/GasStationList.jsx]: Same here

**Review by mbosselmann — APPROVED** [review]
Approved. 🚀 Just two mini comments.

**Pull request merged**